include/picongpu/fields/currentDeposition/Strategy.def

```diff
@@ -40,37 +40,85 @@ namespace picongpu
              * To utilize the device fully you should have enough supercells
              *   - 2D: minimum multiprocessor count * 9 * 4
              *   - 3D: minimum multiprocessor count * 27 * 4
+             *
+             * @{
              */
             struct StridedCachedSupercells
             {
                 static constexpr bool useBlockCache = true;
                 static constexpr bool stridedMapping = true;
                 using BlockReductionOp = nvidia::functors::Atomic<::alpaka::AtomicAdd, ::alpaka::hierarchy::Threads>;
                 using GridReductionOp = nvidia::functors::Add;
+                static constexpr int workerMultiplier = 1;
             };
 
+            /** @tparam T_workerMultiplier Oversubscribe the number of workers used to compute the current by the given
+             * multiplier. Can be used to optimize the device occupancy.
+             */
+            template<int T_workerMultiplier>
+            struct StridedCachedSupercellsScaled
+            {
+                static constexpr bool useBlockCache = true;
+                static constexpr bool stridedMapping = true;
+                using BlockReductionOp = nvidia::functors::Atomic<::alpaka::AtomicAdd, ::alpaka::hierarchy::Threads>;
+                using GridReductionOp = nvidia::functors::Add;
+#if BOOST_COMP_HIP && PIC_COMPUTE_CURRENT_THREAD_LIMITER
+                // HIP-clang creates wrong results if more threads than particles in a frame will be used
+                static constexpr int workerMultiplier = 1;
+#else
+                static constexpr int workerMultiplier = T_workerMultiplier >= 1 ? T_workerMultiplier : 1;
+#endif
```

Comment:
Not a big deal. This piece seems to be present in several places, with these `ifdef` guards. Perhaps there can be a separate constexpr function, or a struct with internal int that expresses this logic, and so in this place it's just `static constexpr int workerMultiplier = newFunction<T_workerMultiplier>();` ?

Reply:
good idea, I will create a constexpr function for this HIP restriction + range correction.